src/Examples/ImmutableCollections.cs

```diff
@@ -253,7 +253,7 @@ static void AssertSequence<T>(IEnumerable<T> expected, IEnumerable<T> actual, st
                 Assert.Null(actual); //, caption);
                 return;
             }
-            if (expected != null) Assert.NotNull(actual); //, caption);
+            Assert.NotNull(actual); //, caption);
 

```

Comment:
Why this change? We only intend to assert that if we don't expect null; what problem are we solving here?

Reply:
We already have a null check above this code, right here:
https://github.com/protobuf-net/protobuf-net/blob/main/src/Examples/ImmutableCollections.cs#L251
this statement is always be true.